transformers/events/DOCUMENTATION.md

```diff
@@ -2,116 +2,206 @@
+* The abi can be found on the `contracts` tab in Etherscan (scroll down).
+You'll need the entire thing as a string, so it's best to use single quotes
+like above.
+* The deployed field is the block number the contract was deployed on. That
+can be found on the Contract's page, by looking for the Contract Creator
+and clicking it's transaction hash. That will take you to the block it was deployed on.
+
+### Create the Transformer
+
+To create a custom transformer you will need to create a new `Transformer`
+struct with a `ToModels` method on it which converts a `core.EventLog` object
+(the raw, untransformed data) to an `event.InsertionModel` (the domain object).
+
+Note that for this step you probably do _not_ need to create the database
+migration. This is because the only thing you'll be saving are addresses, which
+go into the already existing address table. If you do need to create the
+migration, you'll know, because the tests won't pass. The directions here assume
+you do not, yet.
+
+1. Search for the contract on etherscan using it's signature. 
+1. Find the events for the contract in the contract's source.
+1. For each event in the contract:
+    1. Write a test for the event signature in 
+    [`signature_test.go`](../shared/constants/signature_test.go)
+        1. To find out the event's signature you can use this [Keccak-256
+       calculator](https://emn178.github.io/online-tools/keccak_256.html). For Example:
+            * Search Etherscan for the contract `0x39755357759cE0d7f32dC8dC45414CCa409AE24e`. 
+            * Open the 'Contract' tab and in the contract source code and search
+            for the `LogItemUpdate` event. It looks like this: `event LogItemUpdate(uint id);`. 
+            * You can copy that string and paste it into the Keccak-256
+            calculator (link above). Remove the parameter names and use
+            `uint256` for the `unit` so like this: `LogItemUpdate(uint256)`.
+            This will output an event signature.
+            * To verify you got the signature right you can take that signature
+            (in this case
+            `a2c251311b1a7a475913900a2a73dc9789a21b04bc737e050bbc506dd4eb3488`) 
+            and search for it on the events tab of the contract. Make sure you 
+            prefix it with `0x`. The results should include the event you are
+            looking for, provided one already exists for it.
+    1. Make the test pass by updating the function list in `signature.go` as
+       well as updating `method.go` as needed. Both of these files are in the
+       constants package.
+    1. Create test event in [`test_data`](../test_data)
+        * Use Data from a real event in Etherscan wherever possible.
+          Particularly when it comes to Topics and the Data entries. You'll
+          thank me later.
+        * Update constants as needed. Look at older examples of test data for
+          inspiration.
+        * You do not need a database migration at this point.
```

Comment:
I wonder if this bullet point about a migration is necessary yet, since it's already mentioned above that a migration isn't likely necessary on this step.

Reply:
👍